src/test/java/Homework23/BasePage23.java

```diff
@@ -0,0 +1,44 @@
+package Homework23;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+
+public class BasePage23 {
+
+    WebDriver driver;
+    WebDriverWait wait;
+    Actions actions;
+
+    public BasePage23( WebDriver givenDriver) {
+        driver = givenDriver;
+        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+        actions = new Actions(driver);
+        PageFactory.initElements(driver, this);
+    }
+
+    public void click(By locator) {
+        wait.until(ExpectedConditions.elementToBeClickable(locator));
+    }
+
+    public WebElement findElement(By locator) {
+        return wait.until(ExpectedConditions.visibilityOfElementLocated(locator));
+    }
+    public void chooseAllSongsList() {
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("section.music a.songs"))).click();
+    }
```

Comment:
Base Page \ Base Class — Implements the necessary methods for work with webdriver.

We will store the web elements present on each page in the corresponding java class.

The methods `chooseAllSongsList()` must be stored in the `class HomePage`  or `class AllSongPage` because the song is visible on the web site -home page